feature/platform/transceiver/tests/zr_supply_voltage_test/zr_supply_voltage_test.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package zr_supply_voltage_test
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/samplestream"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+)
+
+const (
+	samplingInterval  = 10 * time.Second
+	targetOutputPower = -10
+	targetFrequency   = 193100000
+	intUpdateTime     = 2 * time.Minute
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func verifyVoltageValue(t *testing.T, dut1 *ondatra.DUTDevice, pStream *samplestream.SampleStream[float64], path string) float64 {
+	voltageSample := pStream.Next(t)
+	if voltageSample == nil {
+		t.Fatalf("Voltage telemetry %s was not streamed in the most recent subscription interval", path)
+	}
+	voltageVal, ok := voltageSample.Val()
+	if !ok {
+		t.Fatalf("Voltage %q telemetry is not present", voltageSample)
+	}
+	// Check voltage return value of correct type
+	if reflect.TypeOf(voltageVal).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	}
+	t.Logf("Voltage sample value %s: %v", path, voltageVal)
+	return voltageVal
+}
+
+func TestZrSupplyVoltage(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+
+	for _, port := range []string{"port1", "port2"} {
+		t.Run(fmt.Sprintf("Port:%s", port), func(t *testing.T) {
+			dp := dut.Port(t, port)
+
+			gnmi.Await(t, dut, gnmi.OC().Interface(dp.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+
+			// Derive transceiver names from ports.
+			tr := gnmi.Get(t, dut, gnmi.OC().Interface(dp.Name()).Transceiver().State())
+			component := gnmi.OC().Component(tr)
+
+			outputPower := gnmi.Get(t, dut, component.OpticalChannel().TargetOutputPower().State())
+			if outputPower != targetOutputPower {
+				t.Fatalf("Output power does not match output power, got: %v want :%v", outputPower, targetOutputPower)
+			}
+
+			frequency := gnmi.Get(t, dut, component.OpticalChannel().Frequency().State())
+			if frequency != targetFrequency {
+				t.Fatalf("Frequency does not match frequency, got: %v want :%v", frequency, frequency)
```

Comment:
`t.Fatalf("Frequency does not match target frequency, got: %v want :%v", frequency, targetFrequency)`